paparazzi/src/main/java/app/cash/paparazzi/internal/ImageUtils.kt

```diff
@@ -312,6 +313,36 @@ internal object ImageUtils {
     }
   }
 
+  fun smallestDiffRect(firstImage: BufferedImage, secondImage: BufferedImage): Rectangle? {
+    var rect: Rectangle? = null
+    val width = max(firstImage.width, secondImage.width)
+    val height = max(firstImage.height, secondImage.height)
+
+    for (y in 0 until height) {
+      for (x in 0 until width) {
+        val firstRgb = if (x < firstImage.width && y < firstImage.height) firstImage.getRGB(x, y) else null
```

Comment:
This looks like per-pixel boxing (allocation) on the happy path.

Could it be done with control flow somehow e.g. `continue`?

Reply:
I think it could be improved upon in that if the running dirty width is equal to max width we only need to check the next row up to the first different pixel to modify the 'bottom' and then can skip the rest of the row since it's already going to be included. But in the happy path you will still be required to check all the pixels to be sure the last pixel didn't suddenly change. Can investigate in a follow up